executable-spec/src/Cardano/Ledger/Spec/STS/Update/Data.hs

```diff
@@ -1,29 +1,130 @@
+{-# LANGUAGE DeriveAnyClass #-}
 {-# LANGUAGE DeriveGeneric #-}
 {-# LANGUAGE DerivingVia #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE TypeSynonymInstances #-}
+{-# LANGUAGE FlexibleInstances #-}
 
 module Cardano.Ledger.Spec.STS.Update.Data where
 
-import Data.Word (Word64)
+--import Data.Word (Word64)
 import           Data.Monoid.Generic (GenericMonoid (GenericMonoid),
                      GenericSemigroup (GenericSemigroup))
 import           Data.Set (Set)
 import           GHC.Generics (Generic)
-
+import           Data.Text (Text)
+import           Numeric.Natural (Natural)
+import           Data.Hashable (Hashable)
+import qualified Data.Hashable as H
 import qualified Ledger.Core as Core
+import           Data.Map.Strict (Map)
+
+
+-- |A unique ID of a software update
+newtype UpId = UpId { getUpId :: Integer --Word64
+                    }
+  deriving (Show, Eq, Ord, Hashable, Generic)
+
+instance Core.HasHash (UpId) where
+  hash (UpId i) = Core.Hash $ H.hash i
+
+instance Num UpId where
```

Comment:
This doesn't seem right. I don't expect that we need to perform arithmetic operations on `UpId`'s :thinking:

Reply:
At code line (at Ideation.hs: 156), we use id addition:
```
where
                nextId = maximum $ (Data.UpId 0) : fmap ((Data.UpId 1)+) ids
```

Reply:
Hmmm, in that case I'd expect all we need is an `Enum` instance, and use [`succ`](https://www.stackage.org/haddock/lts-13.30/base-4.12.0.0/Prelude.html#v:succ)  instead.

Reply:
OK I will fix it.

Reply:
I have removed the `id` field from the SIP data type altogether. The `sipHash` field now plays the role of the id.